R/git.R

```diff
@@ -139,3 +139,17 @@ calculate_age <- function(times) {
 convert_unix_to_iso_time <- function(times) {
   strftime(as.POSIXct(times, origin = "1970-01-01", tz = "UTC"))
 }
+
+get_reports <- function(branch, commit, root) {
+  if (branch == "master") {
+    ## Get all reports in commit if on master branch
+    args <- c("ls-tree", "--name-only", "-d", sprintf("%s:src/", commit))
+  } else {
+    ## Get only files which have changed from master copy
+    ## Note this could inclue files as well as directories.
+    ## How can we exclude them?
+    args <- c("diff-tree", "--name-only",
+              sprintf("refs/remotes/origin/master:src/..%s:src/", commit))
```

Comment:
For unmerged branches that can't be fast-forwarded to master I think this is incorrect as I am seeing a lot of reports changed:

```
git diff-tree --name-only refs/remotes/origin/master:src/..origin/vimc-3943:src
```

gives 1 result as wanted

but

```
git diff-tree --name-only refs/remotes/origin/master:src/..origin/vimc-3347_2:src
```

gives 111!

I see better behaviour with

```
git diff --name-only refs/remotes/origin/master...origin/vimc-3347_2 -- src
```

which we can get the relevant things with:

```
git diff --name-only refs/remotes/origin/master...origin/vimc-3347_2 -- src | xargs -I {} dirname {} | sed -E 's;src/([^/]*).*;\1;' | uniq
```

But this is much better done in R with the list from git

Reply:
Not sure what a great test would be here - if you have the stomach for it a little git demo that has divergent branches would be very useful I think

Reply:
Nice one, thanks. Have got a test which recreates that and a fix in now.